src/users/users.controller.ts

```diff
@@ -123,10 +125,28 @@ export class UsersController {
       },
     },
   })
-  async getUserById(@Param('id') id: number): Promise<UserDto> {
+  async getFullUser(@Param('query') query: string): Promise<FullUserDto> {
+    try {
+      this.logger.debug(`Find a full user by email/id: ${query}`);
+      return new FullUserDto(await this.usersService.findUser(query));
+    } catch (err) {
+      throw new HttpException(err.message, err.status);
+    }
+  }
+
+  @Get('/search/:query')
+  @ApiOperation({
+    description: SWAGGER_DESC_FIND_USERS,
+  })
+  @ApiOkResponse({
+    type: UserDto,
+    description: 'Returns all array with user objects',
+  })
+  async searchUsers(@Param('query') query: string): Promise<FullUserDto[]> {
```

Comment:
Why don't use exact query parameters to filter users via the property name such as `GET /users?firstName=Kiril&company=Bright`

Reply:
Request was on feature which will search through all fields

Reply:
@Kif-Fando I don't see any descriptions of the requested changes. There is no GitHub issue, no PR body. Anyway, it doesn't change the fact that the API design is awful.